main.py

```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/env python3
+# This is the Hello-World-Python program ported to Python3 and improved
+# by using the proper enterprise project patterns.
+# Originally written by kmachnicki
+# Disclaimer: Yes, it is a joke :)
 
-#Welcome!
 
-print("Hello World!")
+# This is the heart of the program.
+# This class is responsible for printing the string provided by the caller.
+# It is a proxy for the print function.
+class Printer:
+    # __init__ is responsible for initializing the _printer member, so we can use it later.
+    def __init__(self):
+        self._printer = print
 
-#And that's it ;)
+    # __call__ is used to print the string provided by the caller
+    def __call__(self, string_to_print):

```

Comment:
Every function that accepts or returns some arguments, must have the description formatted the following way:

    Parameters
    ----------
    X : {type}
         Description of the parameter.

    Returns
    -------
    Y : {type}
        Description of returned value.

Please apply this to all other functions.